### PR TITLE
chore: Add type hints for datasets module and improve docs

### PR DESCRIPTION
**Pull request by thatlittleboy**
## Overview

Description of the changes proposed in this pull request:
- Adds some type hints for the datasets in `shap.datasets`
- Standardize / correct some of the output formatting of the docstrings. E.g. we had issues like this:

## Checklist

- [x] All [pre-commit checks](https://pre-commit.com/#install) pass.
- [ ] Unit tests added (if fixing a bug or adding a new feature)

### CODE REVIEW

File: shap/datasets.py [hunk 1]
```diff
@@ -84,10 +102,11 @@ def california(n_points=None):
     To get the processed data and target labels::
 
         data, target = shap.datasets.california()
+
     """
     d = sklearn.datasets.fetch_california_housing()
     df = pd.DataFrame(data=d.data, columns=d.feature_names)
-    target = d.target
+    target: np.ndarray = d.target
```
Comment (connortann) [inline shap/datasets.py:109]: Just to check, should this be `pd.Series`?

**Review by connortann — APPROVED** [review]

**Comment by connortann** [comment]
Looks great! The rendered docs look much better.

**Review by CloseChoice — APPROVED** [review]
Nice work, looks great

File: shap/datasets.py (hunk 1 above)
Comment (CloseChoice) [reply shap/datasets.py:109]: yes, I think it should be a series

Comment (thatlittleboy) [reply shap/datasets.py:109]: @connortann @CloseChoice hi folks, sorry, just to check, do you mean we want to return a pandas series here, or you mean `d.target` is a Series?

Cuz I checked, it's a numpy array:

image

Comment (CloseChoice) [reply shap/datasets.py:109]: I thought `d` is a `pd.Series`. Seems not to be the case so everything's settled in here. Ready to merge IMO.

**Pull request merged**